src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/EnvelopedCms.cs

```diff
@@ -126,6 +126,23 @@ public byte[] Encode()
         // Recipients invoke Decode() to turn the on-the-wire representation into a usable EnvelopedCms instance. Next step is to call Decrypt().
         //
         public void Decode(byte[] encodedMessage)
+        {
+            if (encodedMessage == null)
+                throw new ArgumentNullException(nameof(encodedMessage));
+
+            Decode(new ReadOnlySpan<byte>(encodedMessage));
+        }
+
+        /// <summary>
+        ///   Decodes the provided data as a CMS/PKCS#7 EnvelopedData message.
+        /// </summary>
+        /// <param name="encodedMessage">
+        ///   The data to decode.
+        /// </param>
+        /// <exception cref="CryptographicException">
+        ///   The <paramref name="encodedMessage"/> parameter was not successfully decoded.
+        /// </exception>
+        public void Decode(ReadOnlySpan<byte> encodedMessage)
         {
             if (encodedMessage == null)
```

Comment:
This check is equivalent to `if (encodedMessage._pointer == nullptr && encodedMessage.Length == 0) { /* ... */ }`. Did you intend for these semantics over a normal `IsEmpty` check?

Reply:
Nope, didn't mean to have any check at all (empty hits a different path).  Removed this line, added a test for importing empty.